lib/addons/topics-api.ts

```diff
@@ -0,0 +1,40 @@
+import OptableSDK from "../sdk";
+
+declare module "../sdk" {
+    export interface OptableSDK {
+        tryTopicsAPI: () => void;
+    }
+}
+
+declare global {
+    interface Document {
+        browsingTopics: () => Promise<Array<{
+            configVersion: string;
+            modelVersion: string;
+            taxonomyVersion: string;
+            topic: number;
+            version: string;
+        }>>;
+        featurePolicy:{ allowsFeature: (feature:string) => boolean };
+    }
+}
+
+OptableSDK.prototype.tryTopicsAPI = async function () {
+    if (!sessionStorage.topics_fetched && 'browsingTopics' in document && document.featurePolicy.allowsFeature('browsing-topics')) {
+        const topicsArray = await document.browsingTopics();
+        if (topicsArray.length > 0) {
+            const topics: string[] = [];
+            for (let topic of topicsArray) {
+                topics.push('taxonomy version: ' + topic.taxonomyVersion + ', topic id: ' + topic.topic);
+            }
+
+            if (topics.length > 0) {
+                this.profile({
+                    topics_api: topics.join('|')
+                });
+            }
+        }
+    }
+    sessionStorage.topics_fetched = true;
```

Comment:
What does it involve to always grab the topics from the browser ? I'd rely on caching in last resort as it's usually the source of bugs.

Reply:
I only fetch once per session. Why cache on top of it?

Reply:
What I meant is, why have that flag all together? If you see other similar functions like tryIdentify, if you call it N times, it will do it N times. I'd leave the concern of calling only once to the user of the function.